pkg/generate/app/cmd/newapp.go

```diff
@@ -710,20 +711,146 @@ func (c *AppConfig) Run() (*AppResult, error) {
 	}, nil
 }
 
+// followRefToDockerImage follows a buildconfig...To/From reference until it terminates in docker image information.
+// This can include dereferencing chains of ImageStreamTag references that already exist or which are being created.
+// ref is the reference to To/From to follow. If ref is an ImageStreamTag that is following another ImageStreamTag, isContext
+// should be set to the parent IS. Finally, objects is the list of objects that new-app is creating to support the buildconfig.
+// It returns a reference to a terminal DockerImage or nil if one could not be determined (a valid, non-error outcome). err
+// is only used to indicate that the follow encountered a severe error (e.g malformed data).
+func (c *AppConfig) followRefToDockerImage(ref *kapi.ObjectReference, isContext *imageapi.ImageStream, objects app.Objects) (*kapi.ObjectReference, error) {
+
+	if ref == nil {
+		return nil, fmt.Errorf("Unable to follow nil")
+	}
+
+	if ref.Kind == "DockerImage" {
+		// Make copy so we don't modify any structures new-app/build created
+		copy := *ref
+		// Namespace should not matter here. The DockerImage URL will include project information if it is relevant.
+		copy.Namespace = ""
+
+		// DockerImage names may or may not have a tag suffix. Add
+		// :latest if there is no tag so that string comparison will
+		// behave as expected.
+		if !strings.Contains(copy.Name, ":") {
+			copy.Name += ":" + imageapi.DefaultImageTag
+		}
+		return &copy, nil
+	}
+
+	if ref.Kind != "ImageStreamTag" {
+		return nil, fmt.Errorf("Unable to follow reference type: %q", ref.Kind)
+	}
+
+	isNS := ref.Namespace
+	if len(isNS) == 0 {
+		isNS = c.OriginNamespace
+	}
+
+	// Otherwise, we are tracing an IST reference
```

Comment:
To be clear, this line a verbose guard to let someone know that they are misusing the current method's implementation. e.g. If a new type of To/From ObjectReference is encountered, this path will annoy folks and remind them the cycle check logic needs to be updated -- but do so without breaking newapp.
In short, I don't believe newapp's current implementation can actually trigger the error. I will add a test case to cover the code, but it will need to invoke checkCircularReferences with contrived data.